### PR TITLE
rawspeed IO: handle scanner DNG as non-raw

### LINKED ISSUE #7080: Demosaic/Conversion -> Messages in GUI + Teminal errors (development)
**Describe the bug**

After discarding the histories of all my test files and while it was reassessing all the files the following message showed up in the GUI (lighttable, multiple times):

> history problem detected
> please report via the issue tracker
> include the xmp file

The following was seen on the terminal:

> blockdenom vanishes 
> [dt_dev_pixelpipe_synch] found enabled demosaic in history for non-raw `grainy.II.negative.11.dng`, id: 426
> [dt_ioppr_transform_image_colorspace_cl] invalid conversion from 0 to 2
> [_transform_matrix] invalid conversion from 0 to 2
> [dt_ioppr_transform_image_colorspace] invalid conversion from 0 to 2
> [dt_dev_pixelpipe_synch] found enabled demosaic in history for non-raw `grainy.reflections.dng`, id: 1173
> [_transform_matrix] invalid conversion from 0 to 2
> [dt_ioppr_transform_image_colorspace] invalid conversion from 0 to 2
> blockdenom vanishes 
> [dt_dev_pixelpipe_synch] found enabled demosaic in history for non-raw `villa.bluthner.canon.exr`, id: 1199
> numblox = 0

**To Reproduce**
1. Start darktable from a terminal
2. Select a fim roll/folder that contains exr and/or dng files
3. Select these dng/exr files and discard history
4. See GUI message while darktable is reassessing
5. See errors in terminal

**Expected behavior**
No errors and conversion/demosaicing is done correctly for all file types.

These are clean xmp's to start with, nothing was added/changed by the me. I made sure by removing all the xmps in that specific directory.

**Platform**
* darktable        : 3.3.0+1809 g0415cd920
* OpenCL installed : Yes (nvidia-opencl-common 450.80.02-1)
* OpenCL activated : Yes
* Graphics card    : GeForce RTX 2060 SUPER 7979 MiB
* Graphics driver  : nvidia (450.80.02)
* OS               : Linux (kernel 5.8.10-1)
* Distro           : Debian Buster (rel. 10.6)

**Additional context**
* Self compiled : Yes.
* Compilers     : gcc 8.3.0
* Cmake         : 3.13.4
* Gtk           : 3.24.5-1
* Glib          : 2.58.3-2
* Sqlite        : 3.27.2
* Optimization  : None

**Asked for info**

[grainy.II.negative.11.dng.xmp.txt](https://github.com/darktable-org/darktable/files/5604222/grainy.II.negative.11.dng.xmp.txt)

[grainy.reflections.dng.xmp.txt](https://github.com/darktable-org/darktable/files/5604219/grainy.reflections.dng.xmp.txt)

[villa.bluthner.canon.exr.xmp.txt](https://github.com/darktable-org/darktable/files/5604215/villa.bluthner.canon.exr.xmp.txt)

I'm going to refrain from uploading the RAWs for the moment (26+26+55 Mb ). I'm willing to upload if needed, though.

### PR DESCRIPTION
**Pull request by aurelienpierre**
Should definitely fix #7080 for DNG files produced by scanners.

### CODE REVIEW

File: src/common/imageio_rawspeed.cc [hunk 1]
```diff
@@ -365,18 +370,58 @@ dt_imageio_retval_t dt_imageio_open_rawspeed(dt_image_t *img, const char *filena
...
+    img->flags |= DT_IMAGE_S_RAW;
+
+    if(DataType == TYPE_FLOAT32)
+      img->flags |= DT_IMAGE_HDR;
+    else
+      img->flags &= ~DT_IMAGE_HDR;
+  }
+  else
+  {
+    // Scanners DNG don't need white balance and black point
```
Comment (LebedevRI) [inline src/common/imageio_rawspeed.cc:410]: Citation needed

Comment (aurelienpierre) [reply src/common/imageio_rawspeed.cc:410]: the RGB coeffs are all set to 1. I don't have proof or reference, just example files.

Comment (aurelienpierre) [reply src/common/imageio_rawspeed.cc:410]: another argument is the black point cannot be computed, and even if it was, the Cineon algos rely on a log10 that will fail if anything == 0.

**Comment by aurelienpierre** [comment]
hmm that branch got mixed up, I have cleaning to do.

**Comment by TurboGit** [comment]
@kmilos : Can you comment on this? TIA.

**Comment by kmilos** [comment]
@TurboGit Can't really say much - would be good to have an actual sample exhibiting a specific problem to check if there is still anything needed to be done here...

**Comment by TurboGit** [comment]
So let's close and we will reopen if needed.

**Pull request closed without merging**